### PR TITLE
docker: add pointer to related nix option

### PR DESCRIPTION
**Pull request by Zahrun**
###### Description of changes

Add pointer to related nix option.
Please add backport to 22.11 flag.

@tailhook @vdemeester @periklis @mikroskeem @maxeaubrey 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mikroskeem** [comment]
I think that this pointer doesn't fit into derivation's description, because NixOS modules and packages are separate. I haven't seen this in any other derivations either, it would feel bit out of place compared to others.

**Comment by Zahrun** [comment]
Please see https://github.com/NixOS/nixpkgs/issues/221698 and https://github.com/NixOS/nixpkgs/pull/222397.
I had the idea when I saw [solaar indicating appropriate option in longDescription](https://search.nixos.org/packages?channel=22.11&show=solaar&from=0&size=50&sort=relevance&type=packages&query=solaar).

Actually, I think that it would be even better to add a meta flag relatedOptions that we can fill per package, give all the options that relate to each package and that will pop out visibly in https://search.nixos.org/packages. What do you think?

Maybe it's time to open a separate issue to discuss this.

**Comment by maxeaubrey** [comment]
There is a clear distinction between installing docker (adding the docker package to your enclosure to use as a client when connecting to another host) and enabling the docker service, sockets, and all associated configuration. I think the description as written is misleading. Perhaps discoverability could be somehow improved, but in my opinion this isn't it.

**Comment by Zahrun** [comment]
> There is a clear distinction between installing docker (adding the docker package to your enclosure to use as a client when connecting to another host) and enabling the docker service, sockets, and all associated configuration. I think the description as written is misleading. Perhaps discoverability could be somehow improved, but in my opinion this isn't it.

The option description is "This option enables docker, a daemon that manages linux containers. Users in the "docker" group can interact with the daemon (e.g. to start or stop containers) using the docker command line tool."

So I modified the text accordingly. However, the main idea that I have is that it would be easier if when searching for a package we could directly see the related options. For that maybe the solution of longDescription is not the best.

File: pkgs/applications/virtualization/docker/default.nix [hunk 1]
```diff
@@ -254,6 +254,11 @@ rec {
     meta = with lib; {
       homepage = "https://www.docker.com/";
       description = "An open source project to pack, ship and run any application as a lightweight container";
+      longDescription = ''
+        Docker is a platform designed to help developers build, share, and run modern applications.
+
+        To enable a docker daemon on NixOS, please use the option `virtualisation.docker.enable = true`.
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/docker/default.nix:260]:
```suggestion
        To enable the docker daemon on NixOS, set the `virtualisation.docker.enable` to true.
```

File: pkgs/applications/virtualization/docker/default.nix [hunk 2]
```diff
@@ -254,6 +254,11 @@ rec {
     meta = with lib; {
       homepage = "https://www.docker.com/";
       description = "An open source project to pack, ship and run any application as a lightweight container";
+      longDescription = ''
+        Docker is a platform designed to help developers build, share, and run modern applications.
+
+        To enable the docker daemon on NixOS, set the `virtualisation.docker.enable` to `true`.
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/docker/default.nix:260]:
```suggestion
        To enable the docker daemon on NixOS, set the `virtualisation.docker.enable` option to `true`.
```
what was I thinking

**Pull request merged**